### DISCUSSION TITLE
SCSS module with dynamically added classes

### DISCUSSION THREAD

**Discussion by vladfulgeanu** [vercel/next.js#11398] [Help]
I'm using the 9.3 built-in support for SCSS modules.

The problem that I have is that after rendering (inside `useEffect()`) I want to dynamically add a class to an element using vanilla JS:
`element.classList.add('error')`

But this doesn't work with the current built-in support for SCSS (same for CSS) as it only works when doing something like this:

```
// THIS WORKS
import styles from "./Button.module.scss";

const Button = ({ children }) => (
  <button type="button" className={styles.error}>
    {children}
  </button>
);

export default Button;
```

```
// THIS DOES NOT WORK
import "./Button.module.scss";

const Button = ({ children }) => (
  <button type="button" className="error">
    {children}
  </button>
);

export default Button;
```

Is there a way to make this work with component-level styles or will I have to use global stylesheets?

**Comment by timneutkens** [comment]
`styles.error` refers to the hashed classname generated for the CSS module, meaning that you can use it anywhere, including in this way: `element.classList.add(styles.error)`

**Comment by vladfulgeanu** [comment]
Thanks for the answer, @timneutkens! 

Extending the question even further:
Let's say I have a list of elements, each one having an "item" class set and after it has been rendered, depending on the items' position, I want to add an extra class, either "left" or "right".

The dynamic part will add the classes using 
```
element.classList.add(styles.left) or
element.classList.add(styles.right)
```

JSX looks like this:
```
// Component.jsx
import styles from './Component.module.scss';

{ids.map(id => (
  <div className="item" key={id}> // left & right classes would be added to this <div>
    <div className="arrow" />
    <SmallComponent id={id} />
  </div>
))}
```

And styles module:
```
// Component.module.scss
.left {
  background-color: blue;
}
.right {
  background-color: green;
}
```

The problem that I have appears when making use of sass ampersand and nested styles:
```
.item {
  &.left {
    background-color: blue;
    .arrow {
      height: 20px;
      width: 20px;
      background-color: red;
    }
  }
  &.right {
    background-color: green;
  }
}
```
this simply won't produce the desired effect of applying style definitions to left, right and arrow classes.
I've tried assigning the "item" class this way:
```
...
<div className={styles.item} key={id}>
...
```
but this won't work because I want to be able to query by the "item" name with `document.getElementsByClassName('item')`
So I tried this:
```
...
<div className={`item ${styles.item}`} key={id}>
...
with
element.classList.add('left') or
element.classList.add('right')
```
but still no effect.

Is there anything am I missing? How can I define the nested SCSS classes when they are assigned dynamically?
Thanks very much in advance!

**Reply by bkyam** [reply]
I have the same question. I've resorted to different components for each style. I know that's clunky, but until a better solution presents itself... :/

**Reply by LarsEjaas** [reply]
I think I know how to do this!

```
//this is the dynamic className
const item = 'potato'; 

example:
<div className={Styles.item+ ' ' +Styles[item]} />
```

Please be aware of the syntax inside the {} brackets. The spaces are very important!
